fuel-vm/src/tests/blockchain.rs

```diff
@@ -398,11 +398,165 @@ fn load_external_contract_code() {
+        let expected_gas_used = starting_gas_amount + cost_of_copy + cost_of_noops;
+
+        let actual_gas_used = gas_cost_for_len(&mut client, rng, salt, len);
+        assert_eq!(actual_gas_used, expected_gas_used);
+    }
+}
+
+fn gas_cost_for_len(
+    client: &mut MemoryClient,
+    rng: &mut StdRng,
+    salt: Salt,
+    len: u16,
+) -> Word {
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = Default::default();
+    let height = Default::default();
+
+    // Start by creating and deploying a new example contract
+    let mut contract_code = vec![];
+    for _ in 0..1000 {
+        contract_code.push(op::noop());
+    }
+
+    let bytes = contract_code.into_iter().collect::<Vec<u8>>();
+    let program: Witness = bytes.into();
+
+    let contract = Contract::from(program.as_ref());
+    let contract_root = contract.root();
+    let state_root = Contract::default_state_root();
+    let contract_id = contract.id(&salt, &contract_root, &state_root);
+
+    let input0 = Input::contract(rng.gen(), rng.gen(), rng.gen(), rng.gen(), contract_id);
+    let output0 = Output::contract_created(contract_id, state_root);
+    let output1 = Output::contract(0, rng.gen(), rng.gen());
+
+    let consensus_params = ConsensusParameters::standard();
+
+    let tx_create_target = TransactionBuilder::create(program.clone(), salt, vec![])
+        .gas_price(gas_price)
+        .gas_limit(gas_limit)
+        .maturity(maturity)
+        .add_random_fee_input()
+        .add_output(output0)
+        .finalize()
+        .into_checked(height, &consensus_params)
+        .expect("failed to check tx");
+
+    client.deploy(tx_create_target);
+
+    // Then deploy another contract that attempts to read the first one
+    let reg_a = 0x20;
+    let reg_b = 0x21;
+
+    let count = ContractId::LEN as Immediate12;
+
+    let mut load_contract = vec![
+        op::xor(reg_a, reg_a, reg_a), // r[a] := 0
+        op::ori(reg_a, reg_a, count), // r[a] := r[a] | ContractId::LEN
+        op::aloc(reg_a),              // Reserve space for contract id in the heap
+    ];
+
+    // Generate code for pushing contract id to heap
+    for (i, byte) in contract_id.as_ref().iter().enumerate() {
+        let index = i as Immediate12;
+        let value = *byte as Immediate12;
+        load_contract.extend([
+            op::xor(reg_a, reg_a, reg_a),    // r[a] := 0
+            op::ori(reg_a, reg_a, value),    // r[a] := r[a] | value
+            op::sb(RegId::HP, reg_a, index), // m[$hp+index] := r[a] (=value)
+        ]);
+    }
+
+    // when
+    load_contract.extend([
+        op::move_(reg_a, RegId::HP),  // r[a] := $hp
+        op::xor(reg_b, reg_b, reg_b), // r[b] := 0
+        op::ori(reg_b, reg_b, len),   /* r[b] += 12 (will be
+                                       * padded to 16) */
+        op::ldc(reg_a, RegId::ZERO, reg_b), // Load first two words from the contract
+        op::move_(reg_a, RegId::SSP),       // r[b] := $ssp
```

Comment:
Since this is `reg_a`, should the comment be:

```suggestion
        op::move_(reg_a, RegId::SSP),       // r[a] := $ssp
```